app/models/article.rb

```diff
@@ -53,6 +53,7 @@ class Article < ApplicationRecord
 
   before_validation :evaluate_markdown
   before_validation :create_slug
+  before_validation :check_post_rate_limit
```

Comment:
I don't think it should be checked in a callback, especially not before each time the article gets validated. 

I think `ArticleCreationService` is the right place for `check_post_rate_limit`

Reply:
Thanks for pointing me in the right direction, Ill take another look at it all. Getting my head around a new codebase & language all at once is more difficult than expected!

Reply:
Welcome to the codebase and thanks again! :-)